src/Infrastructure.EntityFramework/Repositories/UserRepository.cs

```diff
@@ -161,6 +161,8 @@ public async Task<Core.Entities.User> GetBySsoUserAsync(string externalId, Guid?
             entity.AccountRevisionDate = user.AccountRevisionDate;
             entity.RevisionDate = user.RevisionDate;
 
+            await dbContext.SaveChangesAsync();
```

Comment:
❓ Why is this here instead of line 180? Doesn't EF handle transactions automagically?

Reply:
From my [understanding](https://www.entityframeworktutorial.net/entityframework6/transaction-in-entity-framework.aspx), this will make the database call for all the previous changes to the user, separating them from all the domain calls below.
CC: @rkac-bw, let me know if I'm wrong here

Reply:
That is my understanding, While the overall transaction makes sure that changes are committed or rolled back as a single unit, the  SaveChangesAsync() within the try block affects when those changes are persisted to the database. Placing it within the try block allows you to make sure that changes are saved immediately instead of stored in change tracking.

Reply:
And it seems that the updateDataActions uses that saved data so it requires the data to be persisted